server/server.py

```diff
@@ -48,16 +55,93 @@ def exec_DB(sql, params=()):
     conn.close()
 
 
+def get_admin_password():
+    result = query_DB("SELECT password FROM users WHERE name='admin'")
+    if result:
+        return result[0][0]
+    else:
+        return None
+
+
+def set_admin_password(admin_password):
+    password_hash = hashlib.sha256()
+    password_hash.update(admin_password)
+    exec_DB("DELETE FROM users WHERE name='admin'")
+    exec_DB("INSERT INTO users VALUES (?, ?, ?)", (None, "admin", password_hash.hexdigest()))
+
+
+def require_admin(func):
+        def wrapper(*args, **kwargs):
+            global session_cookie
+            global last_session_activity
+            global SESSION_TIMEOUT
+            if session_cookie and COOKIE_NAME in cherrypy.request.cookie and session_cookie == cherrypy.request.cookie[COOKIE_NAME].value:
+                if time.time() - last_session_activity > SESSION_TIMEOUT:
+                    return app.disconnect()
```

Comment:
That won't work. You need to `raise cherrypy.HTTPRedirect("/disconnect")` or not use a relative path here: https://github.com/sweetsoftware/Ares/pull/8/files#diff-762988684ae9cb15fd2b07a83e11a4e1R128.
If you simply return the result of `app.disconnect()` the URL won't change and `disconnect` is going to look for `./login` from the current URL.

Example:
- a user is on `/cnc`
- they don't interact with the app for 5+ min (the current session timeout)
- when they try to refresh or whatever they'll see the "you have been disconnected" message but the login link will throw a 404 because it redirects to `/cnc/login`.